src/NexusMods.Library/LibraryService.cs

```diff
@@ -0,0 +1,199 @@
+        _fileExtractor = fileExtractor;
+        _temporaryFileManager = temporaryFileManager;
+
+        _downloadActivitySourceCache
+            .Connect()
+            .Bind(out _downloadActivities)
+            .Subscribe()
+            .DisposeWith(_compositeDisposable);
+
+        _downloadActivitySourceCache
+            .Connect()
+            .WhenPropertyChanged(downloadActivity => downloadActivity.Status)
+            .Where(propertyValue => propertyValue.Value == PersistedDownloadStatus.Completed)
+            .Select(propertyValue => propertyValue.Sender)
+            .SelectMany(AddCompletedDownloadAsync)
+            .Subscribe()
+            .DisposeWith(_compositeDisposable);
+    }
+
+    /// <inheritdoc/>
+    public void EnqueueDownload(IDownloadActivity downloadActivity, bool addPaused = false)
+    {
+        _logger.LogInformation("Adding download `{Title}` to the library", downloadActivity.Title);
+
+        _downloadActivitySourceCache.AddOrUpdate(downloadActivity);
+
+        if (addPaused) return;
+        downloadActivity.Downloader.Start(downloadActivity);
+    }
+
+    private async Task<LibraryFile.ReadOnly> AddCompletedDownloadAsync(
+        IDownloadActivity downloadActivity,
+        CancellationToken cancellationToken)
+    {
+        _downloadActivitySourceCache.Remove(downloadActivity);
+
+        using var tx = _connection.BeginTransaction();
+        var entityId = tx.TempId();
+
+        // TODO: create download specific library item
+        var libraryFile = await AddLibraryFile(tx, entityId, downloadActivity.DownloadPath, cancellationToken: cancellationToken);
+
+        var result = await tx.Commit();
+        return result.Remap(libraryFile);
+    }
+
+    /// <inheritdoc/>
+    public async Task<Optional<LocalFile.ReadOnly>> AddLocalFileAsync(
+        AbsolutePath absolutePath,
+        CancellationToken cancellationToken = default)
+    {
+        _logger.LogInformation("Adding local file at `{Path}` to the library", absolutePath);
+
+        if (!absolutePath.FileExists)
+        {
+            if (absolutePath.DirectoryExists())
+            {
+                _logger.LogError("File at `{Path}` can't be added to the library because it's a directory", absolutePath);
+            }
+
+            _logger.LogError("File at `{Path}` can't be added to the library because it doesn't exist", absolutePath);
+            return Optional<LocalFile.ReadOnly>.None;
+        }
+
+        using var tx = _connection.BeginTransaction();
+        var entityId = tx.TempId();
+
+        var libraryFile = await AddLibraryFile(tx, entityId, absolutePath, cancellationToken: cancellationToken);
+
+        var localFile = new LocalFile.New(tx, entityId)
+        {
+            LibraryFile = libraryFile,
+            OriginalPath = absolutePath.GetFullPath(),
+        };
+
+        var result = await tx.Commit();
+        return result.Remap(localFile);
+    }
+
+    private async Task<LibraryFile.New> AddLibraryFile(
```

Comment:
```suggestion
    private async Task<LibraryFile.New> AddLibraryFileAsync(
```
For naming consistency